### PR TITLE
Add timestamp to all joy messages

### PR DESCRIPTION
**Pull request by lbucklandAS**
Prior to this commit there were some joy messages getting published that were not getting timestamps. This commit adds functionality so that all messages are published with timestamps.

### CODE REVIEW

File: joy/src/joy_node.cpp [hunk 1]
```diff
@@ -291,7 +291,6 @@ class Joystick
             break; // Joystick is probably closed. Definitely occurs.
 
           //ROS_INFO("Read data...");
-          joy_msg.header.stamp = ros::Time().now();
```
Comment (JWhitleyAStuff) [inline joy/src/joy_node.cpp]: Why is this line removed? Isn't this necessary in the case of publish_now being true on line 376 below?

File: joy/src/joy_node.cpp [hunk 2]
```diff
@@ -369,7 +368,6 @@ class Joystick
         }
         else if (tv_set) // Assume that the timer has expired.
         {
-          joy_msg.header.stamp = ros::Time().now();
```
Comment (JWhitleyAStuff) [inline joy/src/joy_node.cpp]: Same here as above.

File: joy/src/joy_node.cpp (hunk 1 above)
Comment (lbucklandAS) [reply joy/src/joy_node.cpp]: @JWhitleyAStuff There's only one point in the code where joy_msg is published, so my thinking was that it made the most sense to update the header timestamp right before publishing in all cases (and other places where the timestamp was updated were redundant). Am I missing something in this approach? Maybe the timestamp works to represent something else other than publish time?

File: joy/src/joy_node.cpp [hunk 3]
```diff
@@ -400,6 +398,7 @@ class Joystick
             }
             pub_.publish(sticky_buttons_joy_msg);
```
Comment (JWhitleyAStuff) [inline joy/src/joy_node.cpp]: If "sticky buttons" is being used, the message is published here. If the header timestamp isn't set higher up, joy_msg.header.stamp won't be populated and will copy 0s to the sticky_buttons_joy_msg.header.stamp.

Comment (lbucklandAS) [reply joy/src/joy_node.cpp]: Ah got it, missed this. I'll clean this up.

**Comment by lbucklandAS** [comment]
@JWhitleyAStuff The issues you pointed out should be resolved.
